homeassistant/components/sensor/ring.py

```diff
@@ -133,5 +143,19 @@ def update(self):
             if history:
                 self._extra = history[0]
                 created_at = self._extra['created_at']
-                self._state = '{0:0>2}:{1:0>2}'.format(
+                raw_state = '{0:0>2}:{1:0>2}'.format(
                     created_at.hour, created_at.minute)
+                if self._clock_format == '12H':
+                    hours, minutes = raw_state.split(':')
```

Comment:
I think you can use `datetime.strptime` instead like the updated answer here:
https://stackoverflow.com/a/13554665